File: frontend/update_sidebar.js
const fs = require('fs');
let content = fs.readFileSync('src/components/Sidebar.jsx', 'utf8');

if (!content.includes('const prefix = userRole')) {
    content = content.replace(
      'const userRole = localStorage.getItem("role");',
      `const userRole = localStorage.getItem("role");\n  const prefix = userRole === "SUPER_ADMIN" ? "/superadmin" : (userRole === "ADMIN" ? "/admin" : (userRole === "HR" ? "/hr" : ""));`
    );

    content = content.replace(/path:\s*\"\/superadmin(\/[a-zA-Z0-9-]+)\"/g, 'path: `${prefix}$1`');
    content = content.replace(/navigate\(\"\/superadmin(\/[a-zA-Z0-9-]+)\"\)/g, 'navigate(`${prefix}$1`)');

    content = content.replace(/path:\s*\"\/admin(\/[a-zA-Z0-9-]+)\"/g, 'path: `${prefix}$1`');
    content = content.replace(/navigate\(\"\/admin(\/[a-zA-Z0-9-]+)\"\)/g, 'navigate(`${prefix}$1`)');

    content = content.replace(/isActive\(\"\/superadmin(\/[a-zA-Z0-9-]+)\"\)/g, 'isActive(`${prefix}$1`)');
    content = content.replace(/isActive\(\"\/admin(\/[a-zA-Z0-9-]+)\"\)/g, 'isActive(`${prefix}$1`)');

    content = content.replace(/navigate\(\"\/superadmin\"\)/g, 'navigate(`${prefix}`)');
    content = content.replace(/navigate\(\"\/admin\"\)/g, 'navigate(`${prefix}`)');
    content = content.replace(/isActive\(\"\/superadmin\"\)/g, 'isActive(`${prefix}`)');
    
    // Also the manual logic checks
    content = content.replace(/location\.pathname === \"\/superadmin\"/g, 'location.pathname === `${prefix}`');
    content = content.replace(/location\.pathname === \"\/admin\"/g, 'location.pathname === `${prefix}`');
    
    
    // There is one tricky line for navigation click in Sidebar.jsx around line 111:
    // if (role === "SUPER_ADMIN" || role === "ADMIN") navigate("/superadmin");
    // else if (role === "HR") navigate("/admin");
    // This will become:
    // if (role === "SUPER_ADMIN" || role === "ADMIN") navigate(prefix);
    // else if (role === "HR") navigate(prefix); // which is basically exactly the prefix!
    
    // I will let it be replaced by navigate(`${prefix}`) which works fine.
    
    fs.writeFileSync('src/components/Sidebar.jsx', content);
    console.log('Sidebar updated');
} else {
    console.log('Sidebar already updated');
}

File: frontend/update_sidebar.py
import re

with open('src/components/Sidebar.jsx', 'r', encoding='utf-8') as f:
    content = f.read()

# Replace the role check to define `prefix`
if 'const prefix = userRole' not in content:
    content = content.replace(
        'const userRole = localStorage.getItem("role");',
        'const userRole = localStorage.getItem("role");\n  const prefix = userRole === "SUPER_ADMIN" ? "/superadmin" : (userRole === "ADMIN" ? "/admin" : (userRole === "HR" ? "/hr" : ""));'
    )

    # Replace /superadmin/ literally inside path mappings and navigate
    content = re.sub(r'path:\s*\"/superadmin(/.*?)\"', r'path: `${prefix}\1`', content)
    content = re.sub(r'navigate\(\"/superadmin(/.*?)\"\)', r'navigate(`${prefix}\1`)', content)
    content = re.sub(r'isActive\(\"/superadmin(/.*?)\"\)', r'isActive(`${prefix}\1`)', content)
    
    # Replace /admin/ literally inside path mappings (for HR)
    content = re.sub(r'path:\s*\"/admin(/.*?)\"', r'path: `${prefix}\1`', content)
    content = re.sub(r'navigate\(\"/admin(/.*?)\"\)', r'navigate(`${prefix}\1`)', content)
    content = re.sub(r'isActive\(\"/admin(/.*?)\"\)', r'isActive(`${prefix}\1`)', content)

    # Replace absolute /superadmin and /admin
    content = re.sub(r'navigate\(\"/superadmin\"\)', r'navigate(`${prefix}`)', content)
    content = re.sub(r'navigate\(\"/admin\"\)', r'navigate(`${prefix}`)', content)
    content = re.sub(r'isActive\(\"/superadmin\"\)', r'isActive(`${prefix}`)', content)
    content = re.sub(r'isActive\(\"/admin\"\)', r'isActive(`${prefix}`)', content)
    
    content = re.sub(r'location\.pathname === \"/superadmin\"', r'location.pathname === `${prefix}`', content)
    content = re.sub(r'location\.pathname === \"/admin\"', r'location.pathname === `${prefix}`', content)

with open('src/components/Sidebar.jsx', 'w', encoding='utf-8') as f:
    f.write(content)

print("done")
